src/main/java/com/elasticbox/jenkins/ElasticBoxSlaveHandler.java

```diff
@@ -266,7 +266,7 @@ private boolean processSubmittedQueue(TaskListener listener) {
                             cloud,
                             "".equals(configDescription) ? config.getId() : configDescription));
                 } else {
-                    if ( !slave.isSingleUse() ) {
+                    if ( slave.isSingleUse() ) {
```

Comment:
Please, add a comment here to clarify that this check is only for the first and second attempts.